### PR TITLE
[RNMobile] Buttons block: Display formatting buttons after closing the block settings (only Android)

### LINKED ISSUE #39491: Buttons block - Opening the settings hides some of the formatting buttons
## Description
While testing another feature a bug was found in the `Buttons` block on **Android** where opening the block's settings would hide some of the formatting options once the Bottom sheet is closed.

Most likely the issue is around [this code](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/button/edit.native.js#L134-L143).

## Step-by-step reproduction instructions
- Open the app
- Add a `Buttons` block
- Open the block's settings
- Dismiss the Bottom sheet

## Expected behaviour
All of the formatting options should be visible after dismissing the Bottom sheet.

## Actual behaviour
Only the link option is visible.

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/4885740/158565487-12b0f671-c306-4587-b965-a3135bb41b7c.mov

## Device information
- Device: Pixel 5 (Emulator)
- Operating system: Android 12
- WordPress app version: Latest

### PR DESCRIPTION
**Pull request by fluiddot**
## What?

This PR fixes an issue on Android related to the Buttons block where the formatting buttons are hidden when opening the block settings and not displayed again after dismissing the settings.

## Why?

Fix https://github.com/WordPress/gutenberg/issues/39491.

## How?

The changes introduced in https://github.com/WordPress/gutenberg/pull/34695 added a workaround for Android to prevent visual glitches when opening the block settings. This is achieved by unfocusing the text input and disabling the button focus when the block settings are opened, which turns to make the formatting buttons to be hidden. This PR updates this logic and checks when any bottom sheet is dismissed to re-enable the button focus and focus the text input again.

## Testing Instructions

**NOTE:** This issue wasn't happening on iOS, so it's recommended to focus the testing on Android.

1. Open a post/page.
1. Add a Buttons block.
1. Observe that the initial Button block is selected.
1. Observe that the formatting buttons in the toolbar are displayed.
1. Tap on the ⚙️ button to open the block settings.
1. Dismiss the bottom sheet.
1. Observe that the formatting buttons in the toolbar are displayed.

## Screenshots or screencast 

iOS|Android
--|--
<video src=https://user-images.githubusercontent.com/14905380/165932252-fc40c07f-7feb-4654-b5a5-a9b8114a710b.mp4>|<video src=https://user-images.githubusercontent.com/14905380/165932442-97e30629-a47a-470d-83de-8256931dfc9e.mp4>

### CODE REVIEW

File: packages/block-library/src/button/edit.native.js [hunk 1]
```diff
@@ -134,10 +134,15 @@ function ButtonEdit( props ) {
 	useEffect( () => {
 		// Blur `RichText` on Android when link settings sheet or button settings sheet is opened,
 		// to avoid flashing caret after closing one of them
-		if ( editorSidebarOpened || isLinkSheetVisible ) {
-			if ( Platform.OS === 'android' && richTextRef?.current ) {
-				richTextRef.current.blur();
+		const isAnyBottomSheetVisible =
```
Comment (jhnstn) [inline packages/block-library/src/button/edit.native.js]: minor consideration: `editorSidebarOpened || isLinkSheetVisible` is fairly easy to grok. I'm wondering if we can drop the `isAnyBottomSheetVisible` const for now. Not a block but thought I'd mention it.

**Review by jhnstn — APPROVED** [review]
Nice clean change! Testing worked as expected.

File: packages/block-library/src/button/edit.native.js (hunk 1 above)
Comment (fluiddot) [reply packages/block-library/src/button/edit.native.js]: Sure, it's only used in a single place so we could use the value directly 👍.

Comment (fluiddot) [reply packages/block-library/src/button/edit.native.js]: I applied this suggestion in https://github.com/WordPress/gutenberg/pull/40725/commits/c1c173cbf7ca8ef375079a3294bd2b016773ef0a.

**Comment by fluiddot** [comment]
> Nice clean change! Testing worked as expected.

Thanks @jhnstn for the review ❤️ ! I'd appreciate it if you could also take a look at the [GB-mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4814), thanks 🙇.

**Pull request merged**